_bleio/common.py

```diff
@@ -450,8 +451,8 @@ def __init__(
         *,
         uuid: UUID,
         properties: int = 0,
-        read_perm: int = Attribute.OPEN,
-        write_perm: int = Attribute.OPEN,
+        read_perm: Attribute = Attribute.OPEN,
+        write_perm: Attribute = Attribute.OPEN,
```

Comment:
I think `int` is correct for these ones. 

It's accessing `Attribute.OPEN` specifically which is defined here as an int (a hexadecimal one, but int none-theless). https://github.com/adafruit/Adafruit_Blinka_bleio/blob/a7d037244956a2d77e99250f76c11f9660d46386/_bleio/attribute.py#L21

Reply:
I may be wrong here, but let me point out that Enum is different in CPython than it is in C (Enum values remain "boxed" in CPython). This is a CPython 3.9.11 session:

```
>>> from enum import Enum
>>> class Attribute(Enum):
...     NO_ACCESS = 0x00
...     OPEN = 0x11
...     ENCRYPT_NO_MITM = 0x21
...     ENCRYPT_WITH_MITM = 0x31
...     LESC_ENCRYPT_WITH_MITM = 0x41
...     SIGNED_NO_MITM = 0x12
...     SIGNED_WITH_MITM = 0x22
... 
>>> Attribute.NO_ACCESS, type(Attribute.NO_ACCESS)
(<Attribute.NO_ACCESS: 0>, <enum 'Attribute'>)
>>> Attribute.NO_ACCESS.value, type(Attribute.NO_ACCESS.value)
(0, <class 'int'>)
>>> isinstance(Attribute.NO_ACCESS, int)
False
>>> isinstance(Attribute.NO_ACCESS, Attribute)
True
```

And, mypy was complaining here. I *think* this is just on CPython. A quick test of `from enum import Enum` on CircuitPython 7.2.3 yielded "no module named 'enum'".

So, relatedly, if you WANT the code to be using the int values we could use the `.value` version. But given that the instance is already taking up space, I don't think we'd save anything there. Maybe a *tiny* bit of compute on comparisons?

Reply:
CircuitPython/MicroPython don't have `enum`; this is just an `int`, as you see. So `Attribute` is in this case just a holder of constants, it's not something that is instantiated. That is deep in the native implementation. So `int` is right.

Reply:
I wrote a rather long thread on discourse as I worked through understanding this myself: https://discordapp.com/channels/327254708534116352/970711335396577310/971144344788676680

The tl;dr would be this question: what do we want mypy to encourage users to type? Right now it says Attribute.NO_ACCESS (i.e., the current default value) is wrong, because that's not an int. The situation is even more confused in Adafruit_CircuitPython_BLE.

Reply:
> The tl;dr would be this question: what do we want mypy to encourage users to type? Right now it says Attribute.NO_ACCESS (i.e., the current default value) is wrong, because that's not an int. The situation is even more confused in Adafruit_CircuitPython_BLE.

This is definitely a good question. I think generally I lean toward encouraging folks to type things as CircuitPython will see them (so `int` in this case) even when it differs from how CPython would interpret it. 

However the Blinka use-case does throw a bit of a wrench into that. Since it's mainly used on Linux with CPython maybe in this case it makes more sense to go toward the CPython way of things? But I am unsure about it's usage with micropython if that behaves the same a CircuitPython with regards to this I could see the case for going with that even though some people will be using it on Linux with CPython.